cmd/gcp-controller-manager/app/node_annotater.go

```diff
@@ -33,15 +33,25 @@ import (
 	corelisters "k8s.io/client-go/listers/core/v1"
 	"k8s.io/client-go/tools/cache"
 	"k8s.io/client-go/util/workqueue"
+	"k8s.io/kubernetes/pkg/apis/core/helper"
 	"k8s.io/kubernetes/pkg/controller"
 
 	"github.com/golang/glog"
 	compute "google.golang.org/api/compute/v1"
 )
 
-const InstanceIDAnnotationKey = "container.googleapis.com/instance_id"
+const (
+	InstanceIDAnnotationKey           = "container.googleapis.com/instance_id"
+	NodeTerminationTaintAnnotationKey = "cloud.google.com/impending-node-termination"
```

Comment:
Is this annotation already set in stone?

Reply:
Annotation not so much - It needs an OSS k8s change which may be tedious - https://github.com/kubernetes/kubernetes/pull/68100/files#diff-71cf2e0d331a511e2025ef83511791a3R54
The taint itself is already being used in the OSS version and so I'd prefer compat if possible.